frontend/src/lib/keybind.ts

```diff
@@ -0,0 +1,246 @@
+/**
+ * Enum to represent the key strings provided by {@link KeyboardEvent}
+ */
+export enum Keys {
+  J = 'j',
+  D = 'd',
+  K = 'k',
+  H = 'h',
+  V = 'v',
+  T = 't',
+  M = 'm',
+  L = 'l',
+  O = 'o',
+  S = 's',
+  SINGLE_QUOTE = "'",
+  FORWARD_SLASH = '/',
+  CTRL = 'Control',
+  ENTER = 'Enter',
+  ARROW_UP = 'ArrowUp',
+  ARROW_DOWN = 'ArrowDown',
+}
+
+// The type used to declare the key downs that will trigger the callback
+export interface KeybindCallback {
+  key: Keys;
+  callback: (e: KeyboardEvent) => void;
+  shift?: boolean;
+  ctrl?: boolean;
+  alt?: boolean;
+}
+
+/**
+ * Used to create and interface with key bindings
+ */
+export class Keybind {
+  // All of the keybindings to their callbacks
+  private bindings: KeybindCallback[];
+  private bindingEntries: Keys[];
+
+  /**
+   * Constructor
+   *
+   * @param bindings An array of {@link KeybindCallback}s to be evaluated and utilized with {@link Keybind:onKeyDown}.
+   */
+  constructor(bindings: KeybindCallback[]) {
+    this.bindings = bindings;
+    this.bindingEntries = this.bindings.map((x) => x.key);
+  }
+
+  /**
+   * Attempt to convert a string to {@link Keys} enum
+   *
+   * @param str - The string to attempt to convert
+   * @returns undefined if unable to convert, otherwise {@link Keys}
+   */
+  private keyEnumFromString(str: string): Keys | undefined {
+    for (const [_, key] of Object.entries(Keys)) {
+      if (key == str) return key;
+    }
+
+    return undefined;
+  }
+
+  /**
+   * Handler for `keydown` events
+   *
+   * @remarks
+   * Requires a wrapper function to pass in `useKeyboardShortcuts` boolean store.
+   *
+   * Will run the given callback if all requirements are met from an item in the
+   * previously given {@link KeybindCallback} array.
+   *
+   * @param e - The `keydown` event of `KeyboardEvent` type
+   * @param useKeyboardShortcuts - A boolean of the user's `useKeyboardShortcuts` preference
+   */
+  onKeyDown(e: KeyboardEvent, useKeyboardShortcuts: boolean) {
+    if (!useKeyboardShortcuts || e.repeat) return;
+
+    const enum_key = this.keyEnumFromString(e.key);
+
+    if (!(enum_key && this.bindingEntries.includes(enum_key))) return;
+
+    // Conditionals to be able to later compare to a potentially undefined binding.shift/.al/.ctrl
+    const shift = e.shiftKey ? true : undefined;
+    const ctrl = e.ctrlKey ? true : undefined;
+    const alt = e.altKey ? true : undefined;
+
+    const binding = this.bindings.find((binding) => binding.key === enum_key);
+    if (binding && binding.alt == alt && binding.shift == shift && binding.ctrl == ctrl) {
+      e.preventDefault();
+      binding.callback(e);
+    }
+  }
+}
+
+/*
+Search callbacks
+*/
+
+/**
+ * Used to indicate what direction {@link navigateResults} should go (up/down)
+ */
+type Direction = 1 | -1;
```

Comment:
```ts
type Direction = "up" | "down";
```
is a bit more descriptive. The type is also pretty self-explanatory and the TsDoc string therefore only clutters it a bit and could be removed